examples/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.10)
+project(opendal-cpp-examples)
+
+set(CMAKE_CXX_STANDARD 17)
+
+include(FetchContent)
+FetchContent_Declare(
+    opendal-cpp
+    URL     https://github.com/apache/incubator-opendal/releases/download/v0.40.0/opendal-cpp-v0.40.0.tar.gz
```

Comment:
Will we have this release?

Reply:
How about point to a commit so that we can run it in CI.

Reply:
I find `SOURCE_SUBDIR` option for cmake. Now we don't need to do extra work to package cpp source code.